mithril-client/Cargo.toml

```diff
@@ -37,6 +37,10 @@ reqwest = { version = "0.12.12", default-features = false, features = [
     "macos-system-configuration",
     "json",
     "stream",
+    "gzip",
+    "zstd",
+    "deflate",
+    "brotli"
```

Comment:
Enabling those features makes our builds heavier so maybe this should be behind one or multiple features ? _(if we do so then those new features should be enabled in the client-cli)_
Usually we let library users decide if they want them or not but maybe we just want to enforce them so compression is always enabled ?